### PR TITLE
Bani/softmax cross entropy with logits NGTF-2428

### PR DESCRIPTION
**Pull request by bani-intelaipg**

### CODE REVIEW

File: ngraph_bridge/ngraph_mark_for_clustering.cc [hunk 1]
```diff
@@ -943,6 +955,10 @@ Status MarkForClustering(Graph* graph, const std::set<string> skip_these_nodes,
       type_constraint_map["Slice"]["Index"] = NGraphIndexDTypes();
       type_constraint_map["Snapshot"]["T"] = NGraphDTypes();
       type_constraint_map["Softmax"]["T"] = NGraphNumericDTypes();
+      // For SoftmaxCrossEntropyWithLogits, see
+      // https://github.com/tensorflow/tensorflow/blob/c95ca05536144451ef78ca6e2c15f0f65ebaaf95/tensorflow/core/ops/nn_ops.cc#L1096
+      type_constraint_map["SoftmaxCrossEntropyWithLogits"]["T"] =
+          NGraphNumericDTypes();
```
Comment (sayantan-nervana) [inline ngraph_bridge/ngraph_mark_for_clustering.cc]:
```suggestion
          NGraphRealDTypes();
```

**Comment by sayantan-nervana** [comment]
We can add some tests here: https://github.com/tensorflow/ngraph-bridge/blob/master/test/python/tensorflow/python_tests_list.txt

File: ngraph_bridge/ngraph_mark_for_clustering.cc (hunk 1 above)
Comment (bani-intelaipg) [reply ngraph_bridge/ngraph_mark_for_clustering.cc]: Agreed. Should we then also update the similar one for  `type_constraint_map["SparseSoftmaxCrossEntropyWithLogits"]["T"] =` ... ? @sayantan-nervana

Comment (sayantan-nervana) [reply ngraph_bridge/ngraph_mark_for_clustering.cc]: Yes.

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/ops.pbtxt#L44100
T-> NGraphRealDTypes
Tlabel-> NGraphIndexDTypes

**Comment by bani-intelaipg** [comment]
@sayantan-nervana  and/or @shresthamalik  please help with a quick review of the last 2 commit changes and then pull in.
